src/DataStructures/Tensor/Tensor.hpp

```diff
@@ -265,6 +265,10 @@ class Tensor<X, Symm, IndexList<Indices...>> {
             Requires<std::conjunction_v<tt::is_tensor_index<N>...>> = nullptr>
   SPECTRE_ALWAYS_INLINE constexpr auto operator()(
       N... /*meta*/) const noexcept {
+    static_assert(tmpl::is_set<N...>::value,
```

Comment:
Maybe we should rename `N` to `TensorIndices` or `TensorIndicesInExpression`? I think that would be a lot easier to read (the bad name is my fault, I know...)

Reply:
Yah, I agree with this change!